packages/composites/bob/src/endpoint/format.ts

```diff
@@ -0,0 +1,39 @@
+import JSONRPC from '@chainlink/json-rpc-adapter'
+import { ExecuteWithConfig } from '@chainlink/types'
+import { Validator, Requester } from '@chainlink/ea-bootstrap'
+import { DEFAULT_RPC_URL, ExtendedConfig } from '../config'
+import { ethers } from 'ethers'
+
+export const NAME = 'format'
+
+export const inputParams = {
+    url: false,
+    chainId: true,
+    blockNumber: true
+}
+
+
+export const execute: ExecuteWithConfig<ExtendedConfig> = async (request, config) => {
+    const validator = new Validator(request, inputParams)
+    if (validator.error) throw validator.error
+    const url = config.RPC_URL || validator.validated.data.url || DEFAULT_RPC_URL
+    const provider = new ethers.providers.JsonRpcProvider(url)
+    const jobRunID = validator.validated.id
+
+    const chainId = validator.validated.data.chainId
+    const blockNumber = validator.validated.data.blockNumber
+
+    const block = await provider.getBlock(blockNumber)
+
+    const response = await JSONRPC.execute({
+      ...request,
+      data: { ...request.data, method: 'eth_getBlockByHash', params: [block.hash, false] },
+    })
+    const coder = new ethers.utils.AbiCoder()
+    response.data.result = coder.encode(
+      ['uint8', 'bytes32', 'bytes32'],
+      [chainId, response.data.result.hash, response.data.result.receiptsRoot]
+    )
+    response.data.result = response.data.result.slice(2)
```

Comment:
It would be nice to type the response to help the code document itself.
Otherwise this slice is pretty opaque.

Reply:
> It would be nice to type the response to help the code document itself.
> Otherwise what this slice is getting pretty opaque.

Sorry I don't really understand what you mean by this, do you mean create a new type for the response data?

Reply:
Yes, exactly, then it will be clear from the code what the `.slice(2)` is.